tools/c7n_oci/tests/conftest.py

```diff
@@ -18,13 +20,20 @@ class CustodianOCITesting(PyTestUtils, OCIFlightRecorder):
     """Pytest OCI Testing Fixture"""
 
 
-@pytest.fixture(scope="function")
+@pytest.fixture(scope=Scope.FUNCTION.value)
 def test(request):
     test_utils = CustodianOCITesting(request)
-    test_utils.addCleanup(reset_session_cache)
     return test_utils
 
 
+@pytest.fixture(scope=Scope.SESSION.value, autouse=True)
+def setup(request):
+    oci_provider = OCI()
+    oci_provider.initialize(Config.empty())
+    yield
```

Comment:
probably should have a try/except around it, to ensure it always cleanups, albeit at session level its not clear its meaningful versus good practice. fwiw, pytest-xdist does session scope once per process.

Reply:
Thanks, I have checked more on this and got your point that using session scope with pytest-xdist can potentially result in memory leaks and impact parallel test execution. My initial idea behind using session scope was to initialising the provider once and clearing the cache at the end of the session to avoid creating the session object for each test.
     To address this issue, I can think of using module or class scope as alternative approaches. I am interested in hearing your opinion on these options and open to exploring other approaches if you have any.

Reply:
the challenge is that session level (once per test process) includes tests from other providers, there's no guaranteed ordering for tests (which in itself would be a potential bug), ie. tests might be run from different providers, that are also doing resets per test. the main goal wrt to tests and session initialization is making it fast and perhaps a no-op, which other providers do sometimes via setting the values such that its not required to do additional api calls in initialize. afaics the only safe layer here is per test without depending on total test ordering.